core/include/gnuradio-4.0/Block.hpp

```diff
@@ -1130,6 +1132,33 @@ class Block : public lifecycle::StateMachine<Derived> {
             },
             inputSpans);
 
+        // non-duplicated, ordered by index, the last Tag (wih max index) wins
+        using InputSpanT = typename gr::PortIn<float>::InputSpan<SpanReleasePolicy::ProcessNone>;
+        using ViewT      = decltype(std::declval<InputSpanT>().tags(0UZ));
+        std::vector<ViewT> allPairViews;
+        allPairViews.reserve(8);
```

Comment:
Note: This local vector can be replaced by an array or even a [concat_view](https://github.com/huixie90/cpp_papers/blob/main/impl/concat/concat.hpp) for all blocks that do not contain a `vector<Port>`. This will be followed up separately from the general logic as discussed.